include/api/debug.h

```diff
@@ -123,14 +123,18 @@ static inline void debug_printTCB(tcb_t *tcb)
 
 static inline void debug_dumpScheduler(void)
 {
-    printf("Dumping all tcbs!\n");
-    printf("Name                                    \tState          \tIP                  \t Prio \t Core%s\n",
-           config_set(CONFIG_KERNEL_MCS) ?  "\t InReleaseQueue" : "");
-    printf("--------------------------------------------------------------------------------------\n");
-    for (tcb_t *curr = NODE_STATE(ksDebugTCBs); curr != NULL; curr = TCB_PTR_DEBUG_PTR(curr)->tcbDebugNext) {
-        debug_printTCB(curr);
+    /* Loop through each core and print out scheduler! */
```

Comment:
Should this take the big kernel log to ensure what is printed is really consistent?

Reply:
That's a good question. I suppose it depends on what you are using it for. If it is just for some simple debugging I don't think you care if the numbers are off by a little bit. If you want to use it to help calculate some sort of performance metrics, then you would probably care. We have really only ever used it for the former.

If anyone cares about the latter, we could look into it.

Reply:
I think for now adding a comment would be sufficient that this is not fully synchronized across multiple cores, but works good enough for debug purposes. The rest could be a follow-up action if it is not trivial.